gradoop-flink/src/main/java/org/gradoop/flink/model/impl/operators/matching/single/cypher/common/pojos/Embedding.java

```diff
@@ -623,4 +654,35 @@ public int hashCode() {
     result = 31 * result + Arrays.hashCode(idListData);
     return result;
   }
+
+  @Override
+  public String toString() {
+    List<List<GradoopId>> idCollection = new ArrayList<>();
+    for (int i = 0; i < size(); i++) {
+      idCollection.add(getIdAsList(i));
+    }
+
+    String idString = idCollection
+      .stream()
+      .map(entry ->
+        {
+          if (entry.size() == 1) {
+            return entry.get(0).toString();
+          } else {
+            return "[" + entry.stream().map(GradoopId::toString).collect(joining(", ")) + "]";
```

Comment:
`joining` is overloaded. you can use something like:
`    String out = Stream.of("foo", "bar").collect(Collectors.joining(", ", "[", "]"));`